std/meta/index.zig

```diff
@@ -414,91 +381,84 @@ test "std.meta.activeTag"
-                        const eql = @field(container_a, field_info.name) 
-                            == @field(container_b, field_info.name);
-                        if(!eql) return false;
-                    },
-                }
+                if(!eql(@field(a, field_info.name), 
+                    @field(b, field_info.name))) return false;
             }
             return true;
         },
+        builtin.TypeId.ErrorUnion =>
+        {
+            if(a) |a_p|
+            {
+                if(b) |b_p| return eql(a_p, b_p) else |_| return false;
+            }
+            else |a_e|
+            {
+                if(b) |_| return false else |b_e| return a_e == b_e;
+            }
+        },
         builtin.TypeId.Union => 
         {
             const info = @typeInfo(T).Union;
             
             if(info.tag_type) |_|
             {
-                const tag_a = activeTag(container_a);
-                const tag_b = activeTag(container_b);
+                const tag_a = activeTag(a);
+                const tag_b = activeTag(b);
                 if(tag_a != tag_b) return false;
                 
                 inline for(info.fields) |field_info|
                 {
                     const enum_field = field_info.enum_field.?;
                     if(enum_field.value == @enumToInt(tag_a))
                     {
-                        switch(@typeId(field_info.field_type))
-                        {
-                            builtin.TypeId.Struct,
-                            builtin.TypeId.Union =>
-                            {
-                                return containerEql(@field(container_a, field_info.name),
-                                    @field(container_b, field_info.name));
-                            },
-                            builtin.TypeId.Array => 
-                            {
-                                return mem.eql(field_info.field_type,
-                                    @field(container_a, field_info.name)[0..],
-                                    @field(container_b, field_info.name)[0..]);
-                            },
-                            else =>
-                            {
-                                return @field(container_a, field_info.name) 
-                                    == @field(container_b, field_info.name);
-                            },
-                        }
+                        return eql(@field(a, enum_field.name),
+                            @field(b, enum_field.name));
                     }
                 }
-                unreachable;
+                return false;
             }
-            @compileError("ContainerEql can only operate on tagged unions.");
+            
+            //This is the only reasonable way to handle an untagged union,
+            // but it will report a false negative in many circumstances.
+            return std.mem.eql(u8, mem.asBytes(&a), mem.asBytes(&b));
```

Comment:
It's probably better if this is a compiler error, to avoid the bugs, where padding bytes are different